### DISCUSSION TITLE
browser bridge?

### DISCUSSION THREAD

**Discussion by ROBERT-MCDOWELL** [SomajitDey/ipfs-chat#12] [Ideas]
would be nice to see a webgui of your chat. with js-ipfs for exemple?

**Comment by SomajitDey** [comment]
Yes, that would indeed be great! `ipfs-chat` can have a flag such as `-s` to launch a server for the webui to connect to. 

What more, if the browser can connect to the main go-ipfs node using websockets, two browsers can then exchange WebRTC signaling data without needing any separate signaling server. After that, there can be video/audio streams too!

But I am afraid this is beyond me, at least for now, because I am a novice at JS. I heartily welcome you and the community to take up the webui projects.

P.S. Decentralizing the IPFS WebRTC transport is currently a [WIP](https://github.com/libp2p/js-libp2p/issues/385). Once it is achieved, I am sure we will be seeing many truly server-less browser-based chat apps.

**Comment by ROBERT-MCDOWELL** [comment]
that's great indeed! for live streaming it would avoid a lot of streams duplications and bandwidth congestion!
thanks for WIP info, I din't know this project.
